db/migrate/20150710081302_create_cloud_ports_and_routers.rb

```diff
@@ -0,0 +1,52 @@
+class CreateCloudPortsAndRouters < ActiveRecord::Migration
+  def change
+    create_table :cloud_ports do |t|
+      t.string     :type
+      t.string     :name
+      t.string     :ems_ref
+      t.belongs_to :ems,               :type => :bigint
+      t.belongs_to :cloud_network,     :type => :bigint
+      t.belongs_to :cloud_subnet,      :type => :bigint
+      t.string     :mac_address
+      t.string     :status
+      t.boolean    :admin_state_up
+      t.string     :device_owner
+      t.string     :device_ref
+      t.integer    :device_id,         :type => :bigint
+      t.string     :device_type
+      t.belongs_to :cloud_tenant,      :type => :bigint
+      t.string     :binding_host_id
+      t.string     :binding_virtual_interface_type
+      t.text       :extra_attributes
+    end
+
+    add_index :cloud_ports, :ems_id
+    add_index :cloud_ports, :cloud_network_id
+    add_index :cloud_ports, :cloud_subnet_id
+    add_index :cloud_ports, [:device_id, :device_type]
+    add_index :cloud_ports, :cloud_tenant_id
+
+    create_table :cloud_ports_security_groups, :id => false do |t|
+      t.belongs_to :cloud_port,        :type => :bigint
+      t.belongs_to :security_group,    :type => :bigint
+    end
+
+    add_index :cloud_ports_security_groups, [:cloud_port_id, :security_group_id], :unique => true, :name => "index_cloud_ports_security_groups"
+
+    create_table :cloud_routers do |t|
+      t.string     :type
```

Comment:
`type` is a scary field name because it usually implies STI and I don't see any STI on this table.  @tenderlove, @matthewd; keep me honest here